content/rdi/quickstart/write-behind-guide.md

```diff
@@ -229,6 +237,105 @@ To use the metrics you can either:
+
+Job definition has the following structure:
+
+```yaml
+source:
+  retries: 1
+  redis:
+    trigger: read-through
+    key_pattern: emp:*
+  keys:
+    delimiter: ":"
+    # alternatively use expression with RegExp
+    fields:
+    - first_name: 1
+    - last_name: 2
+  connection: my-oracle
+  db: test
+  schema: test
+
+  # One of these should be provided:
+  # - table with columns
+  # - sql
+
+  #table: EMP
+  #columns:
+  # - first_name
+  # - last_name
+  # - birth_date
+
+  # bind variables are marked with a colon and must be defined in the `keys` property.
+  sql: |
+    SELECT emp.*, address.*, kids.*
+    FROM emp
+    LEFT JOIN address ON emp.employee_id = address.employee_id
+    LEFT JOIN kids ON emp.employee_id = kids.employee_id
+    WHERE emp.employee_fname= :first_name AND emp.employee_lname = :last_name;
+
+transform:
+  - uses: rename_field
+    with:
+      fields:
+        - from_field: employee_fname
+          to_field: first_name
+        - from_field: employee_lname
+          to_field: last_name
+output:
+  expire: 100
+```
+
+#### The redis section
+
+This section describes the Redis key-pattern and job type:
+
+- The `key_pattern` attribute for the key-pattern that would trigger the read-through job.
+- The `trigger` attribute: indicates it is a read-through and not a write-behind job, using the `read-through` value.
+
+#### The keys section
+
+This section has the following attributes:
+
+- The `delimiter` attribute: specifies a delimiter for tokenizing the Redis key. These tokens are used with the `fields` attribute (see below).
+- Alternatively, specify the `expression` attribute. In this case, the regular expression will be used to tokenize the the Redis key. for example, `expression: "(.+):(\\d+)"` will tokenize the key to anything before the colon and a number after the colon.
+- The `fields` attribute is a map of field name and its offset in the array of tokens generated by the `expression` or the `delimiter` attributes.
+
+#### Downstream database information
+
+The `source` section must specify the following attributes:
+
+- `connection` - A connection alias, referring to connection in the `config.yaml` file.
+- `schema` - The name of the schema to use.
+- `table` or `sql` attributes:
+  - `table` - The table to query.
+  - `columns` - The columns to fetch if `table` is used.
+  - `sql` - A SQL statement in case you would like to `JOIN` several tables or to perform a different complex query. see more below about using this option.
+
+The `source` section can optionally include the following attributes:
+
+- `retries` - Number of attempts to read from the database. The default is 1.
+
+##### Using custom sql queries
+
+You can use any valid `SELECT` statement that can be parsed by the specific downstream database. The values to use in `WHERE` or `ON` sections can be calculated at runtime if you specify one of the keys in the `keys` section.
```

Comment:
maybe explain the binding format with colon.

Reply:
done